lib/traveller.rb

```diff
@@ -95,30 +95,38 @@ def full_state_parse(tokens)
   def check_three_word_states(tokens)
     token_str = tokens.join(' ')
     three_word_states = ['district of columbia']
+    three_word_provinces = ['newfoundland & labrador', 'prince edward island']
+    three_word_states += three_word_provinces
     three_word_states.find { |s| s == token_str }
   end
 
   def check_two_word_states(tokens)
     token_str = tokens.join(' ')
     two_word_states = ['new hampshire', 'new jersey', 'new mexico', 'new york', 'north carolina', 'north dakota', 'puerto rico', 'rhode island', 'south carolina', 'south dakota', 'west virginia' ]
+    two_word_provinces = ['british columbia', 'new brunswick', 'nova scotia', 'northwest territories']
+    two_word_states += two_word_provinces
     two_word_states.find { |s| s == token_str }
   end
 
   def check_one_word_states(tokens)
     token_str = tokens.join(' ')
     one_word_states = ['alabama', 'alaska', 'arizona', 'arkansas', 'california', 'colorado', 'connecticut', 'delaware', 'florida', 'georgia', 'hawaii', 'idaho', 'illinois', 'indiana', 'iowa', 'kansas', 'kentucky', 'louisiana', 'maine', 'maryland', 'massachusetts', 'michigan', 'minnesota', 'mississippi', 'missouri', 'montana',  'nebraska', 'nevada', 'ohio', 'oklahoma', 'oregon', 'pennsylvania', 'tennessee', 'texas', 'utah', 'vermont', 'virginia', 'washington', 'wisconsin', 'wyoming']
+    one_word_provinces = ['alberta', 'manitoba', 'nunavut', 'ontario', 'quebec', 'saskatchewan', 'yukon']
+    one_word_states += one_word_provinces
     one_word_states.find { |s| s == token_str }
   end
 
   def check_abbreviations(tokens)
     token_str = tokens.join(' ')
-    abbreviations = ['al', 'ak', 'az', 'ar', 'ca', 'co', 'ct', 'de', 'dc', 'fl', 'ga', 'hi', 'id', 'il', 'in', 'ia', 'ks', 'ky', 'la', 'me', 'md', 'ma', 'mi', 'mn', 'ms', 'mo', 'mt', 'ne', 'nv', 'nh', 'nj', 'nm', 'ny', 'nc', 'nd', 'oh', 'ok', 'or', 'pa', 'pr', 'ri', 'sc', 'sd', 'tn', 'tx', 'ut', 'vt', 'va', 'wa', 'wv', 'wi', 'wy']
+    state_abbreviations = ['al', 'ak', 'az', 'ar', 'ca', 'co', 'ct', 'de', 'dc', 'fl', 'ga', 'hi', 'id', 'il', 'in', 'ia', 'ks', 'ky', 'la', 'me', 'md', 'ma', 'mi', 'mn', 'ms', 'mo', 'mt', 'ne', 'nv', 'nh', 'nj', 'nm', 'ny', 'nc', 'nd', 'oh', 'ok', 'or', 'pa', 'pr', 'ri', 'sc', 'sd', 'tn', 'tx', 'ut', 'vt', 'va', 'wa', 'wv', 'wi', 'wy']
```

Comment:
You can probably get these from the hash values of us_state_to_abbreviation_mappings and province_to_abbreviation_mappings and not have to duplicate the abbreviations.